File: archive/java/resonantinduction/mechanical/Mechanical.java
package resonantinduction.mechanical;

import net.minecraft.block.Block;
import resonantinduction.core.Reference;
import resonantinduction.core.ResonantInduction;
import resonantinduction.core.Settings;
import resonantinduction.mechanical.belt.BlockConveyorBelt;
import resonantinduction.old.core.recipe.RecipeLoader;
import resonantinduction.old.transport.ResonantInductionTransport;
import calclavia.lib.network.PacketHandler;
import cpw.mods.fml.common.Mod;
import cpw.mods.fml.common.Mod.EventHandler;
import cpw.mods.fml.common.Mod.Instance;
import cpw.mods.fml.common.ModMetadata;
import cpw.mods.fml.common.SidedProxy;
import cpw.mods.fml.common.event.FMLInitializationEvent;
import cpw.mods.fml.common.event.FMLPreInitializationEvent;
import cpw.mods.fml.common.network.NetworkMod;
import cpw.mods.fml.common.network.NetworkRegistry;
import dark.lib.CoreRegistry;

/**
 * Resonant Induction Archaic Module
 * 
 * @author DarkCow, Calclavia
 * 
 */
@Mod(modid = Mechanical.ID, name = Mechanical.NAME, version = Reference.VERSION, dependencies = "required-after:" + ResonantInduction.ID)
@NetworkMod(channels = Reference.CHANNEL, clientSideRequired = true, serverSideRequired = false, packetHandler = PacketHandler.class)
public class Mechanical
{
	/** Mod Information */
	public static final String ID = "ResonantInduction|Mechanical";
	public static final String NAME = Reference.NAME + " Mechanical";

	@Instance(ID)
	public static Mechanical INSTANCE;

	@SidedProxy(clientSide = "resonantinduction.mechanical.ClientProxy", serverSide = "resonantinduction.mechanical.CommonProxy")
	public static CommonProxy proxy;

	@Mod.Metadata(ID)
	public static ModMetadata metadata;

	public static Block blockConveyorBelt;

	@EventHandler
	public void preInit(FMLPreInitializationEvent evt)
	{
		NetworkRegistry.instance().registerGuiHandler(this, proxy);
		blockConveyorBelt = CoreRegistry.createNewBlock("RIConveyorBelt", ID, BlockConveyorBelt.class);
		proxy.preInit();
	}

	@EventHandler
	public void init(FMLInitializationEvent evt)
	{
		Settings.setModMetadata(metadata, ID, NAME);
		proxy.init();
	}
}

File: src/main/scala/resonantinduction/mechanical/Mechanical.scala
package resonantinduction.mechanical

import cpw.mods.fml.common.Mod.EventHandler
import cpw.mods.fml.common.event.{FMLInitializationEvent, FMLPostInitializationEvent, FMLPreInitializationEvent}
import cpw.mods.fml.common.network.NetworkRegistry
import cpw.mods.fml.common.{Mod, ModMetadata, SidedProxy}
import net.minecraft.item.ItemStack
import net.minecraftforge.common.MinecraftForge
import resonant.engine.content.debug.TileCreativeBuilder
import resonant.lib.loadable.LoadableHandler
import resonant.lib.network.discriminator.PacketAnnotationManager
import resonant.lib.schematic.SchematicPlate
import resonantinduction.core.interfaces.IMechanicalNode
import resonantinduction.core.{Reference, ResonantPartFactory, ResonantTab}
import resonantinduction.mechanical.fluid.pipe.PartPipe
import resonantinduction.mechanical.mech.MechanicalNode
import resonantinduction.mechanical.mech.gear.PartGear
import resonantinduction.mechanical.mech.gearshaft.PartGearShaft
import resonantinduction.mechanical.mech.turbine._
import universalelectricity.api.core.grid.NodeRegistry

/**
 * Resonant Induction Mechanical Module
 *
 * @author DarkCow, Calclavia
 */
@Mod(modid = Mechanical.ID, name = "Resonant Induction Mechanical", version = "", modLanguage = "scala", dependencies = "before:ThermalExpansion;required-after:ResonantInductionCore;after:ResonantInduction|Archaic") object Mechanical
{
    /** Mod Information */
    final val ID: String = "ResonantInduction|Mechanical"
    final val NAME: String = Reference.name + " Mechanical"

    @SidedProxy(clientSide = "resonantinduction.mechanical.ClientProxy", serverSide = "resonantinduction.mechanical.CommonProxy")
    var proxy: CommonProxy = null

    @Mod.Metadata(ID)
    var metadata: ModMetadata = null

    val loadables = new LoadableHandler

    @EventHandler
    def preInit(evt: FMLPreInitializationEvent)
    {
        NetworkRegistry.INSTANCE.registerGuiHandler(this, proxy)
        MinecraftForge.EVENT_BUS.register(new MicroblockHighlightHandler)

        loadables.applyModule(proxy)
        loadables.applyModule(MechContent)

        TileCreativeBuilder.register(new SchematicPlate("schematic.waterTurbine.name", MechContent.blockWaterTurbine))
        TileCreativeBuilder.register(new SchematicPlate("schematic.windTurbine.name", MechContent.blockWindTurbine))
        TileCreativeBuilder.register(new SchematicPlate("schematic.electricTurbine.name", MechContent.blockElectricTurbine))

        NodeRegistry.register(classOf[IMechanicalNode], classOf[MechanicalNode])

        ResonantTab.itemStack(new ItemStack(MechContent.blockGrinderWheel))

        PacketAnnotationManager.INSTANCE.register(classOf[TileWindTurbine])
        PacketAnnotationManager.INSTANCE.register(classOf[TileWaterTurbine])

        ResonantPartFactory.register(classOf[PartGear])
        ResonantPartFactory.register(classOf[PartGearShaft])
        ResonantPartFactory.register(classOf[PartPipe])

        loadables.preInit()
    }

    @EventHandler
    def init(evt: FMLInitializationEvent)
    {
        loadables.init()
    }

    @EventHandler
    def postInit(evt: FMLPostInitializationEvent)
    {
        loadables.postInit()
    }
}